components/azure.key.manager/src/main/java/org/wso2/azure/client/AzureADClient.java

```diff
@@ -0,0 +1,332 @@
+        if (client != null) {
+            try {
+                appClient.deleteApplication(client.getId());
+            } catch (KeyManagerClientException e) {
+                e.printStackTrace();
+                handleException("Error occurred while deleting Azure AD Application", e);
+            }
+        }
+    }
+
+    private ClientInformation getClientInformationByClientId(String clientId) throws APIManagementException {
+        ClientInformation client = null;
+        try {
+            ClientInformationList list = appClient.searchByAppId(clientId);
+            if (list != null && list.getValue().size() > 0)
+                client = list.getValue().get(0);
+        } catch (KeyManagerClientException e1) {
+            e1.printStackTrace();
+            handleException("Error occurred while searching Azure AD Application", e1);
+        }
+        return client;
+    }
+
+    @Override
+    public OAuthApplicationInfo retrieveApplication(String clientId) throws APIManagementException {
+        ClientInformation client = getClientInformationByClientId(clientId);
+        if (client != null) {
+            return this.getOAuthApplicationInfo(client);
+        }
+        return null;
+    }
+
+    @Override
+    public AccessTokenInfo getNewApplicationAccessToken(AccessTokenRequest tokenRequest) throws APIManagementException {
+        TokenGenerator tokenGenerator = new TokenGenerator();
+        String clientId = tokenRequest.getClientId();
+        String clientSecret = tokenRequest.getClientSecret();
+
+        return tokenGenerator.getAccessTokenInfo(clientId, clientSecret, tokenEndpoint);
+    }
+
+    protected void clearTest() throws KeyManagerClientException {
+        ClientInformationList allTestApplications = this.appClient.getAllTestApplications();
+        allTestApplications.getValue().forEach(v -> {
+            try {
+                this.appClient.deleteApplication(v.getId());
+            } catch (KeyManagerClientException e) {
+                e.printStackTrace();
+            }
+        });
+    }
+
+    @Override
+    public KeyManagerConfiguration getKeyManagerConfiguration() throws APIManagementException {
+        return this.configuration;
+    }
+
+    @Override
+    public OAuthApplicationInfo mapOAuthApplication(OAuthAppRequest oAuthAppRequest) throws APIManagementException {
+        String consumerKey = oAuthAppRequest.getOAuthApplicationInfo().getClientId();
+        String consumerSecret = oAuthAppRequest.getOAuthApplicationInfo().getClientSecret();
+
+        if (StringUtils.isNotBlank(consumerKey)) {
+            OAuthApplicationInfo clientInfo = retrieveApplication(consumerKey);
+            if (clientInfo == null) {
+                String msg = "Something went wrong while getting OAuth application for given consumer key "
+                        + consumerKey;
+                log.error(msg);
+                throw new APIManagementException(msg);
+            }
```

Comment:
Let not log and throw. 
```suggestion
                throw new APIManagementException(msg);
            }
```